extended/src/main/java/io/kubernetes/client/extended/controller/builder/ControllerWatchBuilder.java

```diff
@@ -0,0 +1,98 @@
+package io.kubernetes.client.extended.controller.builder;
+
+import io.kubernetes.client.extended.controller.Controllers;
+import io.kubernetes.client.extended.controller.DefaultControllerWatch;
+import io.kubernetes.client.extended.controller.reconciler.Request;
+import io.kubernetes.client.extended.workqueue.WorkQueue;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public class ControllerWatchBuilder<ApiType> {
+
+  private Function<ApiType, Request> workKeyGenerator;
+  private WorkQueue<Request> workQueue;
+  private Class<ApiType> apiTypeClass;
+
+  private Predicate<ApiType> onAddFilterPredicate;
+  private BiPredicate<ApiType, ApiType> onUpdateFilterPredicate;
+  private BiPredicate<ApiType, Boolean> onDeleteFilterPredicate;
+
+  ControllerWatchBuilder(Class<ApiType> apiTypeClass, WorkQueue<Request> workQueue) {
+    this.workQueue = null; // required, or throw an IllegalStateException in the end.
```

Comment:
Why is this? That doesn't seem to make sense...

Reply:
removed